cmd/chat_app_socket_server/main.go

```diff
@@ -0,0 +1,47 @@
+// main.go
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	runtime "github.com/Hammad887/chat-app"
+	"github.com/Hammad887/chat-app/handlers"
+	request "github.com/gorilla/handlers"
+)
+
+func main() {
+	ctx := context.Background()
+
+	router := mux.NewRouter()
+	// Set up CORS handling
+	headers := request.AllowedHeaders([]string{"*"})
+	origins := request.AllowedOrigins([]string{"*"}) // Allow requests from any origin
+
+	// Add CORS middleware
+	router.Use(request.CORS(headers, origins))
+
+	rt, err := runtime.NewRuntime()
+	if err != nil {
+		log.Println(err)
+	}
+
+	service := rt.Service()
+
+	router.HandleFunc("/ws", func(w http.ResponseWriter, r *http.Request) {
+		err = handlers.WebSocketHandler(w, r, service, ctx)
+		if err != nil {
+			log.Println(err)
+		}
+	})
+
+	fmt.Println("Server is running on port 8085")
+	err = http.ListenAndServe(":8085", router)
```

Comment:
its better to use env variable for port number

Reply:
Done